ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/SQSTest.kt

```diff
@@ -0,0 +1,261 @@
+                    fillSingleTextField(fifoQueueName.substringBefore(".fifo"))
+                    find<ComponentFixture>(byXpath("//div[@accessiblename='FIFO']")).click()
+                    pressCreate()
+                    client.waitForCreation(fifoQueueName)
+                }
+            }
+            val queueUrl = client.getQueueUrl { it.queueName(queueName) }.queueUrl()
+            step("Expand SQS node") { awsExplorer { expandExplorerNode(sqsNodeLabel) } }
+            step("Standard queue") {
+                openSendMessagePane(queueName)
+                step("Send a message and validate it is sent") {
+                    fillSingleJBTextArea("message")
+                    findAndClick("//div[@text='Send']")
+                    // Make sure it shows a sent message
+                    findByXpath("//div[contains(@accessiblename, 'Sent message ID')]")
+                }
+                step("Add 10 messages to the queue so poll will be guaranteed to work") {
+                    client.sendMessageBatch {
+                        it.queueUrl(queueUrl).entries(
+                            (1..10).map { num ->
+                                SendMessageBatchRequestEntry.builder().messageBody("bmessage$num").id(num.toString()).build()
+                            }
+                        )
+                    }
+                }
+                openPollMessagePane(queueName)
+                step("View messages") {
+                    findAndClick("//div[@accessiblename='View Messages' and @class='JButton']")
+                    // Wait for the table to be populated (It's very fast for small queues)
+                    Thread.sleep(1000)
+                    find<JTreeFixture>(byXpath("//div[@class='TableView']")).findAllText().any { it.text.contains("bmessage") }
+                }
+            }
+            closeToolWindowTab()
+            step("FIFO queue") {
+                openSendMessagePane(fifoQueueName)
+                step("Send a message and validate it is sent") {
+                    // fill the message box
+                    fillSingleJBTextArea("message")
+                    // Fill the rest of the fields (deduplication and group ids)
+                    fillAllJBTextFields("message")
+                    findAndClick("//div[@text='Send']")
+                    // Make sure it shows a sent message
+                    findByXpath("//div[contains(@accessiblename, 'Sent message ID')]")
+                }
+            }
+            closeToolWindowTab()
+            step("edit queue settings") {
+                awsExplorer {
+                    openExplorerActionMenu(sqsNodeLabel, queueName)
+                }
+                findAndClick("//div[@text='$editQueueAttributesAction']")
+                // Make sure the dialog opens, then close it
+                find<NewProjectWizardDialog>(DialogFixture.byTitle(editQueueAttributesTitle)).close()
```

Comment:
So we don't actually mutate any of the attributes?

Reply:
I can add that, the reason I did not is because it's a big pain since they are all the same type of element (so they aren't easy to distinguish). I might go with a fill all method, let me see what works.